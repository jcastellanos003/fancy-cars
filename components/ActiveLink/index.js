import { useRouter } from 'next/router'
import Link from 'next/link';

const ActiveLink = ({ children, ...props }) => {
  const router = useRouter();
  const child = React.Children.only(children);
  
  return (
    <Link {...props}>
      {React.cloneElement(child, { active: router.pathname === props.href, theme: props.theme })}
    </Link>
  )
};

export default ActiveLink;
