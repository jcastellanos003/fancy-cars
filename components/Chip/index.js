import PropTypes from 'prop-types';

import * as components from './styles';

export const Chip = ({ name, onClose }) => (
  <components.StyledChip>
    <span>{name}</span> <components.CloseIcon onClick={onClose} />
  </components.StyledChip>
);

Chip.propTypes = {
  name: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired
};
