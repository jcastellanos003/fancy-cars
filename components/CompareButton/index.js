import * as components from './styles';

export const CompareButton = () => (
  <components.StyledCompareButton>
    <components.CompareButtonIcon />
  </components.StyledCompareButton>
);
