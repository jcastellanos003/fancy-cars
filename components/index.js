export * from './ActiveLink';
export * from './Nav';
export * from './Meta';
export * from './SearchBox';
export * from './Card';
export * from './VehicleCard';
export * from './BackButton';
export * from './PageHeader';
export * from './Chip';
export * from './CompareButton';
export * from './PageTitle';
export * from './PriceLabel';
