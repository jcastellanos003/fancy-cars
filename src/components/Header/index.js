import { Nav } from '../Nav';
import * as styled from './styles';

export const Header = () => (
  <section>

    <styled.CustomHeader>
      <styled.Logo src="/logo.png"/>
    </styled.CustomHeader>

    <Nav />

  </section>
);
