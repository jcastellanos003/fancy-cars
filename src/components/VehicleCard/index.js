import PropTypes from 'prop-types';

import { Card, CardHeader, CardContent  } from '../Card';
import { VehicleImage } from './styles';

export const VehicleCard = ({
    image,
    size,
    children
  }) => (
    <section>
      <Card size={size}>

        <CardHeader>
          <VehicleImage src={image} />
        </CardHeader>

        <CardContent>
          {children}
        </CardContent>

      </Card>
    </section>
);

VehicleCard.propTypes = {
  size: PropTypes.oneOf(['medium', 'large']).isRequired,
  image: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};
