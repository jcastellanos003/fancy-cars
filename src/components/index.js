export * from './ActiveLink/ActiveLink';
export * from './BackButton/BackButton';
export * from './Card';
export * from './CompareButton/CompareButton';
export * from './CompareLink/CompareLink';
export * from './Chip/Chip';
export * from './Header/Header';
export * from './HorizontalPane/HorizontalPane';
export * from './InnerPage/InnerPage';
export * from './Master/Master';
export * from './Meta/Meta';
export * from './Nav/Nav';
export * from './PageHeader/PageHeader';
export * from './PageTitle/PageTitle';
export * from './PriceLabel/PriceLabel';
export * from './SearchBox/SearchBox';
export * from './SelectedCard/SelectedCard';
export * from './SelectedButton/SelectedButton';
export * from './VehicleCard/VehicleCard';
