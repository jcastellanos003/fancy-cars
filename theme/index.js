export * from './DefaultTheme';
export * from './GlobalStyles';
