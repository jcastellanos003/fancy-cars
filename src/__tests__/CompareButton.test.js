import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components'

import { CompareButton } from '../components';
import { theme } from '../theme';

describe('CompareButton Component', () => {
  test('Matches the snapshot', () => {
    const handleClick = event => event.preventDefault();
    const id = 1;
    const compareButton = create(
      <ThemeProvider theme={theme}>
        <CompareButton onClick={handleClick} id={id} />
      </ThemeProvider>
      );

    expect(compareButton.toJSON()).toMatchSnapshot();
  })
})
