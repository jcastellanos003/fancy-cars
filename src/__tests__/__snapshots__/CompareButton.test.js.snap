// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareButton Component Matches the snapshot 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  color: #2A293B;
  width: 22px;
  height: 22px;
}

.c0 {
  background: #FAFAFA;
  border-radius: 50%;
  cursor: pointer;
  width: 30px;
  height: 30px;
  position: absolute;
  top: 0.5rem;
  right: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 11px 40px 0 rgba(0,0,0,0.25),0 2px 10px 0 rgba(0,0,0,0.12);
  -webkit-transition: -webkit-transform .2s;
  -webkit-transition: transform .2s;
  transition: transform .2s;
}

.c0 span::after {
  content: '';
  display: block;
  top: 0;
  left: 0;
  width: 30px;
  height: 30px;
  position: absolute;
  border-radius: 50%;
  background: transparent;
}

.c0:hover {
  background: #FFC82C;
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
}

<div
  className="c0"
  onClick={[Function]}
>
  <span
    item={1}
  >
    <svg
      aria-hidden="true"
      className="c1 c2"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <path
          d="M0 0h24v24H0V0z"
        />
      </defs>
      <clipPath>
        <use
          overflow="visible"
          xlinkHref="#a"
        />
      </clipPath>
      <path
        clipPath="url(#b)"
        d="M9.01 14H2v2h7.01v3L13 15l-3.99-4v3zm5.98-1v-3H22V8h-7.01V5L11 9l3.99 4z"
      />
    </svg>
  </span>
</div>
`;
